### DISCUSSION TITLE
Configuring multiple alertmanager urls in Thanos

### DISCUSSION THREAD

**Discussion by ajaikumarsadasivam** [thanos-io/thanos#4461] [Questions & Answers]
We are trying to configure Thanos to send alerts to 1+ more alertmanagers.
As per the configurations: https://thanos.io/tip/components/rule.md/#flags - we tried to follow the config parameter .i.e.,  
_**--alertmanagers.url=ALERTMANAGERS.URL ...  Alertmanager replica URLs to push firing alerts. Ruler claims success if push to at least one alertmanager from discovered succeeds. The scheme should not be empty e.g `http` might be used. The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect Alertmanager IPs through respective DNS lookups. The port defaults to 9093 or the SRV record's value. The URL path is used as a prefix for the regular Alertmanager API path.**_

When we configure as below:
```
- --alertmanagers.url=['https://abc-alertmanager.com/','https://xyz-alertmanager.com/']
- --alertmanagers.url=["https://abc-alertmanager.com/","https://xyz-alertmanager.com/"]
- --alertmanagers.url={'https://abc-alertmanager.com/','https://xyz-alertmanager.com/'}
```

For any of the above combinations, we get below errors:
`level=error ts=2021-07-20T15:27:33.415669197Z caller=rule.go:379 msg="refreshing alertmanagers failed" err="parse URL \"[\\\"https://abc-alertmanager.com/\\\",\\\"https://xyz-alertmanager.com/\\\"]\": parse [\"https://abc-alertmanager.com/\",\"https://xyz-alertmanager.com/\"]: first path segment in URL cannot contain colon"`

Please provide sample configurations on how we can configure multiple urls for the above parameter

**Comment by fktkrt** [comment]
Hey, you can specify multiple Alertmanagers by repeating the argument multiple times with different URLs as per https://github.com/thanos-io/thanos/pull/661/files

E.g. `--alertmanagers.url="http://alertmanager1.domain.com:9093" --alertmanagers.url="http://alertmanager2.domain.com:9093"`
It looks like it's missing from the current version of the Rule docs though.

Based on this, it's also possible by using `--alertmanagers.config ` and `--alertmanagers.config-file` as well:
https://thanos.io/tip/components/rule.md/#alertmanager
